src/components/AbsenceList/test.tsx

```diff
@@ -65,4 +103,92 @@ describe("AbsenceList", () => {
     expect(rahafDate).toBeInTheDocument();
     expect(enyaDate).toBeInTheDocument();
   });
+
+  describe("Sort options", () => {
+    test("Can sort by name", async () => {
+      (fetchAbsences as jest.Mock).mockReturnValue(mockAbsenceData);
+      const user = userEvent.setup();
+      render(
+        <QueryClientProvider client={mockQueryClient}>
+          <AbsenceList />
+        </QueryClientProvider>
+      );
+      const table = await screen.findByRole("table");
+      const { findAllByRole } = within(table);
+      await user.click(screen.getByText("Name"));
+      let items = await findAllByRole("row");
+      expect(items[1]).toHaveTextContent("Enya");
+      await user.click(screen.getByText("Name"));
+      items = await findAllByRole("row");
+      expect(items[1]).toHaveTextContent("Raniya");
+    });
+    test("Can sort by absence type", async () => {
+      (fetchAbsences as jest.Mock).mockReturnValue(mockAbsenceData);
+      const user = userEvent.setup();
+      render(
+        <QueryClientProvider client={mockQueryClient}>
+          <AbsenceList />
+        </QueryClientProvider>
+      );
+      const table = await screen.findByRole("table");
+      const { findAllByRole } = within(table);
+      await user.click(screen.getByText("Absence Type"));
+      let items = await findAllByRole("row");
```

Comment:
so you can use findBy, which has waitFor built in. So when we know the content is already on the page (ie loading has already happened), you could use getBy, as that will instantly get it. You don't need to change this, just an FYI, as I can see why you have used findBy when getting the table, to check that loading has finished.